app/org/sagebionetworks/bridge/validators/StudyValidator.java

```diff
@@ -143,7 +143,20 @@ public void validate(Object obj, Errors errors) {
         validateEmails(errors, study.getTechnicalEmail(), "technicalEmail");
         validateEmails(errors, study.getConsentNotificationEmail(), "consentNotificationEmail");
         validateDataGroupNamesAndFitForSynapseExport(errors, study.getDataGroups());
-        
+
+        // emailVerificationEnabled=true (public study):
+        //     externalIdValidationEnabled and externalIdRequiredOnSignup can vary independently
+        // emailVerificationEnabled=false:
+        //     externalIdValidationEnabled and externalIdRequiredOnSignup must both be true
```

Comment:
This is probably not 100% true now that you can have phone number accounts. In that case we're assuming that the phone is not verified until you successfully use it to sign in, and that's the behavior we're seeking with these flags for email-based studies.

Reply:
That's technically true but (a) we have code that keys off of account status and email verification and nothing currently looks at phone verification and (b) this change is meant to be a quick change so we can patch the email verification vulnerability, not an exhaustive feature change.